File: processual_api/static/js/adapters/cgt.js
const CGT_ADAPTER = (() => {
  async function evaluate(data) {
    return CLIENT.post('/cgt/evaluate', {
      transition_channel: data.transition_channel,
      compatibility: data.compatibility, retention: data.retention,
      harmony: data.harmony, fatigue: data.fatigue,
      complexity: data.complexity, shock: data.shock,
      dwell_time: data.dwell_time, carrier: data.carrier,
      diversity: data.diversity, novelty: data.novelty,
      lift: data.lift
    });
  }

  return { evaluate };
})();

File: processual_api/schemas/cgt.py
from pydantic import BaseModel


class FateView(BaseModel):
    stability: float
    hybridity: float
    distortion: float
    extinction: float
    collapse: float
    flourishing: float
    rank: str
    rank_label_ar: str = ""
    recommendation: str = ""


class CGTEvaluateRequest(BaseModel):
    transition_channel: float = 0.5
    compatibility: float = 0.5
    retention: float = 0.5
    harmony: float = 0.5
    fatigue: float = 0.3
    complexity: float = 0.3
    shock: float = 0.1
    dwell_time: float = 3.0
    carrier: float = 0.5
    diversity: float = 0.5
    novelty: float = 0.5
    lift: float = 0.0


class CGTEvaluateResponse(BaseModel):
    fate_vector: dict[str, float]
    existence_rank: str
    view: FateView | None = None
